src/index.ts

```diff
@@ -64,7 +67,9 @@ export class DynamoDBAutoIncrement {
         await this.props.doc.get({
           AttributesToGet: [this.props.counterTableAttributeName],
           Key: this.props.counterTableKey,
-          TableName: this.props.counterTableName,
+          TableName: this.props.counterTableCopyItem
+            ? this.props.tableName
+            : this.props.counterTableName,
```

Comment:
For the history tables that would contain the item copies, I don't think it makes sense to store the Latest version in the history table, since it would be a duplicate of the data in the main table. 

For circulars, the idea being that the `circulars` table has some item:
```
{
  'circularId': 12345,
  'version': 2,
  'subject': ...,
  'body': ...
}
```
This is the latest version, and is about to be updated to:
```
{
  'circularId': 12345,
  'version': {# to be determined},
  'subject': ...,
  'body': ...
}
```

The history table only contains the old versions, so just:
```
{
  'circularId': 12345,
  'version': 1,
  'subject': ...,
  'body': ...
}
``` 
So the latest value is the `version` attribute from the `circulars` table

Reply:
I need to think about this a bit. Storing all versions including the latest in the history table has the advantage that it will simplify querying the versions of a document: only one query is required. Storing all versions _except_ the latest in the history table has the advantage of requiring half as much storage, in the limit that most documents have only one version.

Reply:
Yea, it seems our comments in this and below are heading to the same point. I had settled on the latter for this design